spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/mapping/MappingRuleInterpreterGroupExplode.scala

```diff
@@ -40,29 +40,63 @@ case class MappingRuleInterpreterGroupExplode(rule: MappingConformanceRule,
       array(rule.attributeMappings.values.toSeq.map(arrCol(_).cast(StringType)): _*),
       typedLit(mappings))
 
-    val placedDf = ExplodeTools.nestedRenameReplace(joined, rule.outputColumn, rule.outputColumn)
+    val withErrorsDf = if (rule.additionalColumns.isEmpty) {
+      val joined = explodedDf.as(CommonMappingRuleInterpreter.inputDfAlias)
+        .join(mapTable.as(CommonMappingRuleInterpreter.mappingTableAlias), joinCondition, CommonMappingRuleInterpreter.joinType)
+        .select(col(s"${CommonMappingRuleInterpreter.inputDfAlias}.*"),
+          col(s"${CommonMappingRuleInterpreter.mappingTableAlias}.${rule.targetAttribute}") as rule.outputColumn)
+
+      val placedDf = ExplodeTools.nestedRenameReplace(joined, rule.outputColumn, rule.outputColumn)
 
-    val arrayErrorCondition = col(rule.outputColumn).isNull.and(expCtx.getArrayErrorCondition(rule.outputColumn))
-    log.debug(s"Array Error Condition = $arrayErrorCondition")
-    val errorsDf = addErrorsToErrCol(placedDf, rule.outputColumn, defaultValueOpt, mappingErrUdfCall, arrayErrorCondition)
+      val arrayErrorCondition = col(rule.outputColumn).isNull.and(expCtx.getArrayErrorCondition(rule.outputColumn))
+      log.debug(s"Array Error Condition = $arrayErrorCondition")
+      addErrorsAndDefaults(placedDf, rule.outputColumn, defaultValuesMap.get(rule.targetAttribute), mappingErrUdfCall, arrayErrorCondition)
+    } else {
+      val outputElements = rule.allOutputColumns().map { case (outputColumn: String, targetAttribute: String) =>
+        val newOutputColName = if (outputColumn.contains(".")) outputColumn.split("\\.").last else outputColumn
+        col(s"${CommonMappingRuleInterpreter.mappingTableAlias}.$targetAttribute") as newOutputColName
+      }.toSeq
+
+      val outputsStructColumnName = if (rule.outputColumn.contains(".")) parentPath + "." + getOutputsStructColumnName(df) else getOutputsStructColumnName(df)
+      val columns = Seq(col(s"${CommonMappingRuleInterpreter.inputDfAlias}.*"),
+        struct(outputElements: _*) as outputsStructColumnName)
+
+      val joined = explodedDf.as(CommonMappingRuleInterpreter.inputDfAlias)
+        .join(mapTable.as(CommonMappingRuleInterpreter.mappingTableAlias), joinCondition, CommonMappingRuleInterpreter.joinType)
+        .select(columns: _*)
+
+      val placedDf = ExplodeTools.nestedRenameReplace(joined, outputsStructColumnName, outputsStructColumnName)
+      val arrayErrorCondition = getMultipleErrorCondition(expCtx, outputsStructColumnName)
+
+      log.debug(s"Array Error Condition = $arrayErrorCondition")
+      if(parentPath=="") {
+        flattenOutputsAndAddErrorsAndDefaultsInFlat(outputsStructColumnName, placedDf, defaultValuesMap,
+          mappingErrUdfCall, arrayErrorCondition)
+      } else flattenOutputsAndAddErrorsAndDefaultsInNested(placedDf, parentPath,
+        outputsStructColumnName, defaultValuesMap, mappingErrUdfCall, arrayErrorCondition)
+    }
```

Comment:
This method's code does not seem particularly readable and maintainable. Perhaps it could get split somehow to apply different LoDs. Even though all the individual parts make sense, together it's just seems to solve too many things at once.